Build error when following https://microsoft.github.io/CCF/main/contribute/build_ccf.html guide on Codespaces

Fresh Codespace

```shell
root@codespaces-9f2b24:/workspaces/CCF# uname -a
Linux codespaces-9f2b24 5.4.0-1094-azure #100~18.04.1-Ubuntu SMP Mon Oct 17 11:44:30 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
root@codespaces-9f2b24:/workspaces/CCF# mkdir build
root@codespaces-9f2b24:/workspaces/CCF# cd build/
root@codespaces-9f2b24:/workspaces/CCF/build# cmake -GNinja -DCOMPILE_TARGET=virtual ..
-- Setting build type to 'RelWithDebInfo' as none was specified.
-- Found Git: /usr/bin/git (found version "2.25.1") 
-- The C compiler identification is Clang 10.0.0
-- The CXX compiler identification is Clang 10.0.0
-- Check for working C compiler: /usr/bin/clang-10
-- Check for working C compiler: /usr/bin/clang-10 -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/clang++-10
-- Check for working CXX compiler: /usr/bin/clang++-10 -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- CCF version = ccf-3.0.0-rc1-62-ge8adbf946
-- CCF release version = 3.0.0
-- CCF version suffix = rc1-62-ge8adbf946
-- Debian package version: 3.0.0~rc1-62-ge8adbf946
-- CMAKE_INSTALL_PREFIX is '/opt/ccf_virtual'
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- The ASM compiler identification is Clang
-- Found assembler: /usr/bin/clang-10
-- Compile target platform: virtual
-- Looking for crypto library - found
-- Looking for dl library - found
CMake Warning at cmake/common.cmake:12 (_message):
  -- Looking for sgx_dcap_ql library - not found.  Attestations based on
  quotes would not function without the quote provider.
Call Stack (most recent call first):
  /opt/openenclave/lib/openenclave/cmake/openenclave-config.cmake:122 (message)
  cmake/open_enclave.cmake:9 (find_package)
  cmake/ccf_app.cmake:22 (include)
  cmake/common.cmake:88 (include)
  CMakeLists.txt:49 (include)


-- Performing Test OE_SPECTRE_MITIGATION_C_FLAGS_SUPPORTED
-- Performing Test OE_SPECTRE_MITIGATION_C_FLAGS_SUPPORTED - Success
-- Performing Test OE_SPECTRE_MITIGATION_CXX_FLAGS_SUPPORTED
-- Performing Test OE_SPECTRE_MITIGATION_CXX_FLAGS_SUPPORTED - Success
-- QuickJS prefix: /workspaces/CCF/3rdparty/exported/quickjs version: 2021-03-27
-- Using sss at /workspaces/CCF/3rdparty/internal/sss
-- Found OpenSSL: /usr/lib/x86_64-linux-gnu/libcrypto.so (found version "1.1.1f")  
-- Performing Test CONST_QUALIFIED_MALLOC_USABLE_SIZE
-- Performing Test CONST_QUALIFIED_MALLOC_USABLE_SIZE - Failed
-- Found protobuf compiler: /opt/protoc/bin/protoc
-- Configuring done
-- Generating done
-- Build files have been written to: /workspaces/CCF/build
root@codespaces-9f2b24:/workspaces/CCF/build# ninja
[53/468] Building CXX object CMakeFiles/ccf.virtual.dir/src/enclave/main.cpp.o
FAILED: CMakeFiles/ccf.virtual.dir/src/enclave/main.cpp.o 
/usr/bin/clang++-10  -DCCF_LOGGER_NO_DEPRECATE -DHAVE_ARPA_INET_H=1 -DINSIDE_ENCLAVE -DNGHTTP2_STATICLIB -DOE_BUILD_HOST_VERIFY -DPLATFORM_VIRTUAL -DVIRTUAL_ENCLAVE -D_FORTIFY_SOURCE=2 -D_LIBCPP_HAS_THREAD_API_PTHREAD -I../include -I../src -I../3rdparty/exported/quickjs -I../3rdparty/exported/QCBOR -I../3rdparty/exported/t_cose/inc -I../3rdparty/exported/nghttp2/includes -isystem ../3rdparty/exported -isystem ../3rdparty/internal -isystem generated -isystem /opt/openenclave/include -O2 -g -DNDEBUG -fPIC   -stdlib=libc++ -Wall -Wextra -Werror -Wundef -Wpedantic -Wno-unused -Wno-unused-parameter -DCONFIG_VERSION=\"2021-03-27\" -DCONFIG_BIGNUM -fstack-protector-strong -std=gnu++2a -MD -MT CMakeFiles/ccf.virtual.dir/src/enclave/main.cpp.o -MF CMakeFiles/ccf.virtual.dir/src/enclave/main.cpp.o.d -o CMakeFiles/ccf.virtual.dir/src/enclave/main.cpp.o -c ../src/enclave/main.cpp
Killed
[58/468] Building CXX object CMakeFiles/node_frontend_test.dir/src/node/rpc/test/node_frontend_test.cpp.o
ninja: build stopped: subcommand failed.
root@codespaces-9f2b24:/workspaces/CCF/build# 


```

--- a/doc/contribute/build_ccf.rst
+++ b/doc/contribute/build_ccf.rst
@@ -33,6 +33,10 @@ Alternatively, on a non-SGX machine, you can build a `virtual` instance of CCF:
 
     CCF defaults to building in the `RelWithDebInfo <https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html>`_ configuration.
 
+.. warning::
+
+    A machine with at least 32Gb of memory is recommended to build CCF with the default compiler.
+
 Build Switches
 --------------
 

